### ISSUE TITLE
Issue when upgrading julia 

### ISSUE THREAD

**Issue by ptiede** [johnnychen94/jill.py#39]
I am trying to use jill to upgrade to julia 1.5.1 using jill install --upgrade and am getting an error. The traceback is
 ```python
Traceback (most recent call last):
  File "/home/ptiede/.local/bin/jill", line 8, in <module>
    sys.exit(main())
  File "/home/ptiede/.local/lib/python3.8/site-packages/jill/__main__.py", line 15, in main
    fire.Fire({
  File "/home/ptiede/.local/lib/python3.8/site-packages/fire/core.py", line 138, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/home/ptiede/.local/lib/python3.8/site-packages/fire/core.py", line 463, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
  File "/home/ptiede/.local/lib/python3.8/site-packages/fire/core.py", line 672, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
  File "/home/ptiede/.local/lib/python3.8/site-packages/jill/install.py", line 372, in install_julia
    installer(package_path, install_dir, symlink_dir, version, upgrade)
  File "/home/ptiede/.local/lib/python3.8/site-packages/jill/install.py", line 201, in install_julia_linux
    copy_root_project(version)
  File "/home/ptiede/.local/lib/python3.8/site-packages/jill/install.py", line 160, in copy_root_project
    old_ver = last_julia_version(version)
  File "/home/ptiede/.local/lib/python3.8/site-packages/jill/install.py", line 77, in last_julia_version
    proj_versions = sorted(filter(lambda ver: sort_key(ver) < version,
  File "/home/ptiede/.local/lib/python3.8/site-packages/jill/install.py", line 77, in <lambda>
    proj_versions = sorted(filter(lambda ver: sort_key(ver) < version,
  File "/home/ptiede/.local/lib/python3.8/site-packages/jill/install.py", line 72, in sort_key
    return float(ver.lstrip("v"))
ValueError: could not convert string to float: '1.5.bak'
````

It looks like the version isn't formatted as you expect which is causing the conversion from string to float to fail. 

I believe I am using version 0.7.0 of jill and python 3.8

**Comment by johnnychen94** [comment]
Thanks for the report!

There's only one thing that `--upgrade` flag does: if the installed version is a newer one (say v1.5 vs v1.4), then `cp -r ~/.julia/environments/v1.4 ~/.julia/environment/v1.5` This isn't very useful in practice and would be confusing, I should give an `experimental` flag to this. (Cref: https://github.com/abelsiqueira/jill/issues/22)

If it's upgrading from Julia `1.5.0` to `1.5.1`, then no other flag is needed. `jill install 1.5.1` would just get you there.

**Comment by johnnychen94** [comment]
I actually never tried this myself in practice and I feel I should deprecate this flag. Any thoughts?

**Comment by johnnychen94** [comment]
It turns out an oversight that I should filter out the `v1.5.bak` folder you created manually. The patch is released with v0.7.1 now.

**Comment by ptiede** [comment]
Oh perfect! Thanks
